Migrating the rest of (non perf) Gemma3 tests to CIv2 from CIv1

mstojkoTT opened the pull request:
### Ticket
[Link to Github Issue
](https://github.com/tenstorrent/tt-metal/issues/30457)

### Problem description
- Before this PR we had 1 Gemma3 test migrated to CIv2. This PR makes all other (non perf) tests migrate also.
- Reason for migrating only non perf tests - for now CIv2 does not support perf tests (because it's using VMs so it's not deterministic)

### What's changed
This PR is 2 changes basically:
1. Putting other tests into CIv2
2. Changing a Siglip test so it's more in line with Gemma tests (with how it gets weight path)
3. Something independtent from this CI2 changes:
- updated the targets for Gemma3 vision perf (`models/demos/gemma3/tests/perf_targets/targets_test_perf_vision_cross_attention_transformer.json`)
- making that same test from above more robust
- some hotfix for sliding attention in `models/tt_transformers/tests/test_decoder_prefill.py` (more details in comment)

### Tests
https://github.com/tenstorrent/tt-metal/actions/runs/18937942949
https://github.com/tenstorrent/tt-metal/actions/runs/18937935010

mstojkoTT committed:
cherry of /moved to ci/

mstojkoTT committed:
Siglip changes

mstojkoTT committed:
empty - fixed test - models/demos/gemma3/tests/test_perf_vision_cross_attention_transformer.py

mstojkoTT commented on models/tt_transformers/tt/model_config.py:
```diff
@@ -1616,6 +1616,9 @@ def _set_params_from_dict(self, config, is_hf=False):
 
         self.query_pre_attn_scalar = text_config.get("query_pre_attn_scalar", None)
 
+        # Sliding window attention
```
This has been already adressed in https://github.com/tenstorrent/tt-metal/pull/31218/files#diff-25ae939c24cd7ee859b9833f30faaf9b9a39833ce7e410e68407bc738c5d9552 and should be merged into main soon.

mstojkoTT commented on models/demos/gemma3/tests/test_ci_dispatch.py:
```diff
@@ -43,21 +40,24 @@ def test_ci_dispatch(hf_model_name, is_ci_env, is_ci_v2_env, model_location_gene
         "models/tt_transformers/tests/test_mlp.py",
-        # "models/tt_transformers/tests/test_attention.py",
-        # "models/tt_transformers/tests/test_attention_prefill.py",
+        "models/tt_transformers/tests/test_attention.py",
```
Already adressed in previous PR that is about to merge https://github.com/tenstorrent/tt-metal/pull/31218

handrewsTT commented on models/tt_transformers/tests/test_decoder_prefill.py:
```diff
@@ -177,13 +177,19 @@ def test_decoder_inference(
-        attn_mask = torch.full((max_seq_len, max_seq_len), torch.finfo(torch.float32).min)
-        attn_mask_torch = torch.triu(attn_mask, diagonal=1)
+        attn_mask = torch.triu(torch.full((max_seq_len, max_seq_len), torch.finfo(torch.float32).min), diagonal=1)
+        if model_args.sliding_window > 0 and (
```
No it won't, python resolves the conditions in order, if `model_args.sliding_window` is `None` it won't try the comparison `model_args.sliding_window > 0`

mstojkoTT committed:
temp

mstojkoTT committed:
temp - added new targets and updated test

mstojkoTT committed:
total new targets, now run

mstojkoTT committed:
disabled SAVE_NEW_PERF_TARGETS

mstojkoTT committed:
allow empty - fixed

mstojkoTT committed:
removed script for updating tests

mstojkoTT committed:
fixed targets

ppetrovicTT commented on a.sh:
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
careful with unwanted files

mstojkoTT committed:
removed a.sh

mstojkoTT committed:
updated target for N300 4b bcs its failing on CI

mstojkoTT committed:
ALL TESTS PASSING. More details below

N300 gemma4b
	    "models/tt_transformers/tests/test_decoder_prefill.py",

            "models/tt_transformers/tests/test_attention.py",
            "models/tt_transformers/tests/test_decoder.py",
            "models/tt_transformers/tests/test_rms_norm.py",
            "models/demos/gemma3/tests/test_vision_transformer_block.py",

            "models/tt_transformers/tests/test_attention_prefill.py",

            "models/demos/gemma3/tests/test_vision_attention.py",

            "models/demos/gemma3/tests/test_mmp.py",

            // "models/demos/gemma3/tests/test_perf_vision_cross_attention_transformer.py",
            "models/demos/siglip/tests/test_attention.py",
            "models/demos/gemma3/tests/test_patch_embedding.py",
            "models/demos/gemma3/tests/test_vision_cross_attention_transformer.py",
            "models/demos/gemma3/tests/test_vision_embedding.py",
            "models/demos/gemma3/tests/test_vision_layernorm.py",
            "models/demos/gemma3/tests/test_vision_mlp.py",
            "models/demos/gemma3/tests/test_vision_pipeline.py",
            // # "models/demos/gemma3/tests/test_vision_rmsnorm.py",
            "models/demos/gemma3/tests/test_vision_transformer.py",
            "models/tt_transformers/tests/test_embedding.py",
            "models/tt_transformers/tests/test_mlp.py",

T3K 27b:
        "models/tt_transformers/tests/test_decoder_prefill.py",O
	"HF_MODEL": "meta-llama/Llama-3.2-3B-Instruct",u
